swift/Tests/Blockchains/OntologyTests.swift

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import TrustWalletCore
+import XCTest
+
+extension String {
+    public func indexOf<T>(stuff: T) -> Int where T: StringProtocol {
+        let r = range(of: stuff)
+        guard let rr = r else {
+            return -1
+        }
+        return rr.lowerBound.encodedOffset
+    }
+}
+
+class OntologyTests: XCTestCase {
+    
+    func testSignOngBalanceOf() {
+        
+        let input = TWOntologySigningInput.with {
+            $0.contract = "ONG".data(using: .utf8)!
+            $0.method = "balanceOf".data(using: .utf8)!
+            $0.queryAddress = "ANDfjwrUroaVtvBguDtrWKRMyxFwvVwnZD".data(using: .utf8)!
+        }
+        
+        let output = OntologySigner.sign(input: input)
+        let result = output.encoded.hexString
+        
+        XCTAssertEqual(86, result.indexOf(stuff: "1446b1a18af6b7c9f8a4602f9f73eeb3030f0c29b70962616c616e63654f661400000000000000000000000000000000000000020068164f6e746f6c6f67792e4e61746976652e496e766f6b65"))
```

Comment:
why not just compare two hex string?

Reply:
Because the `nonce `and `signature` in transaction is random. So, we jump these value to provide a deterministic unit test.

Reply:
I construct a valid tx and check it equals to the rawtx in `TransactionTests.cpp`, maybe this unit test satisfy our needs. https://github.com/NashMiao/wallet-core/blob/master/tests/Ontology/TransactionTests.cpp